scipy/integrate/_quadrature.py

```diff
@@ -141,9 +144,9 @@ def trapezoid(y, x=None, dx=1.0, axis=-1):
         ret = (d * (y[tuple(slice1)] + y[tuple(slice2)]) / 2.0).sum(axis)
     except ValueError:
         # Operations didn't work, cast to ndarray
-        d = np.asarray(d)
-        y = np.asarray(y)
-        ret = np.add.reduce(d * (y[tuple(slice1)]+y[tuple(slice2)])/2.0, axis)
+        d = xp.asarray(d)
+        y = xp.asarray(y)
+        ret = xp.add.reduce(d * (y[tuple(slice1)]+y[tuple(slice2)])/2.0, axis)
```

Comment:
`add.reduce` isn't in the standard, I think that syntax is specific to NumPy ufuncs. I think we can replace it here with `xp.sum` though (which will also add along `axis`: https://data-apis.org/array-api/latest/API_specification/generated/array_api.sum.html).